kubernetes/cache/cache.go

```diff
@@ -178,11 +192,27 @@ func (c *kialiCacheImpl) CheckNamespace(namespace string) bool {
 	return c.isKubernetesSynced(namespace) && c.isIstioSynced(namespace)
 }
 
+// RefreshNamespace will delete the specific namespace's cache and create a new one.
+func (c *kialiCacheImpl) RefreshNamespace(namespace string) {
+	if nsChan, exist := c.stopChan[namespace]; exist {
```

Comment:
I think the close and deletion of the stopChannel should be protected by the lock.